apps/client/src/features/app-settings/panel/sources-panel/import-map/ImportMapForm.tsx

```diff
@@ -77,7 +77,15 @@ export default function ImportMapForm(props: ImportMapFormProps) {
   const isLoading = Boolean(loading);
   const canSubmitSpreadsheet = isSpreadsheet && !isLoading;
   const canSubmitGSheet = !isLoading && !stepData.worksheet.error;
-  const canSubmit = !hasErrors && isValid && (canSubmitSpreadsheet || canSubmitGSheet);
+  const canSubmit = isValid && (canSubmitSpreadsheet || canSubmitGSheet);
+
+  // Set first sheet as default worksheet when 'event schedule' sheet is not there 
```

Comment:
suggestion: lets move the `useEffect` to the top before the derived values, it makes it easier to spot and avoids mistakes with conditional hooks